### PR TITLE
Build base tarball

### PR DESCRIPTION
**Pull request by dseefeld**
Add generation of a base tarball.  This is one that has no artifacts for reference-packages, previously source-built packages and dotnet SDK.  Add a `prep.sh` script to allow prepping the tarball to be run on a given platform.  Also add generation of base tarball in source-build CI.

### CODE REVIEW

File: support/tarball/prep.sh [hunk 1]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
```
Comment (omajid) [inline support/tarball/prep.sh:1]: This file has a permission of `0644`. It should be `0755` (executable by everyone), right?

File: build-source-tarball.sh [hunk 2]
```diff
@@ -98,6 +104,11 @@ while :; do
     shift
 done
 
+# if [ $BASE_TARBALL -eq 1 ] && [[ '$CUSTOM_SDK_DIR' != ''] || [ '$CUSTOM_REF_PACKAGES_DIR' != '' ] || [ '$CUSTOM_PREVIOUSLY_BUILT_PACKAGES_DIR' != '' ]]; then
```
Comment (MichaelSimons) [inline build-source-tarball.sh:107]: Why is this commented out?

File: support/tarball/prep.sh [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            (cd $SCRIPT_ROOT/packages/archive/ && curl -O $line)
+        fi
+    fi
+done < $SCRIPT_ROOT/packages/archive/archiveArtifacts.txt
+
+# Check for the version of dotnet to install
+if [ "$installDotnet" == "true" ]; then
+    echo "  Installing dotnet..."
+    (source ./eng/common/tools.sh && InitializeDotNetCli true)
+fi
```
Comment (MichaelSimons) [inline support/tarball/prep.sh:89]: Missing EOF

**Review by MichaelSimons — COMMENTED** [review]
Please link to an issue that provides the context as to why this change is being made.  I'd like to see what the overall execution plan is.  For example what are the plans for removing the rid specific tarballs and migrating this generic tarball?  With the base tarball how will CI change?  Will there be a single leg that creates the tarball and then it gets tested on various rids?  What is the scope of this initial PR?

File: support/tarball/prep.sh (hunk 1 above)
Comment (dseefeld) [reply support/tarball/prep.sh:1]: Thanks.  Fixed.

**Comment by dseefeld** [comment]
See https://github.com/dotnet/source-build/issues/2143 for an implementation plan for rolling out base tarballs.

File: .vsts.pipelines/jobs/ci-linux.yml [hunk 4]
```diff
@@ -121,6 +122,21 @@ jobs:
     timeoutInMinutes: 45
     continueOnError: true
 
+  # Create base tarball.
```
[the remark this answers is not in the document]
Comment (dseefeld) [reply .vsts.pipelines/jobs/ci-linux.yml:125]: I agree.  I think there'll be a point in time where there'll be both, but I'd like to move towards only using base tarballs. I think I'll leave as is.

File: support/tarball/prep.sh (hunk 3 above)
Comment (dseefeld) [reply support/tarball/prep.sh]: Fixed

File: build-source-tarball.sh (hunk 2 above)
Comment (dseefeld) [reply build-source-tarball.sh]: This was a WIP that I didn't finish.  It should not allow building a base tarball with the --with* parameters.  I've fixed it.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**